pytest_splunk_addon/splunk.py

```diff
@@ -541,52 +621,234 @@ def uf_external(request):
+                    if service_port == "8089":
+                        if "exposed_uf_ports.log" in file:
+                            os.environ["port_uf"] = exposed_port
+                        else:
+                            os.environ["port"] = exposed_port
+                    if service_port == "9997":
+                        os.environ["port_s2s"] = exposed_port
+                    if service_port == "514":
+                        os.environ["sc4s_port"] = exposed_port
+    except Exception as e:
+        LOGGER.error("Exception occurred while port-forwarding")
+
+
+def update_k8s_manifest_files(folder, file):
+    try:
+        with open(
+            os.path.join("{0}".format(folder), "{0}.yaml".format(file)), "r"
+        ) as deployment_file:
+            file_updated = os.path.expandvars(deployment_file.read())
+        with open(
+            os.path.join("{0}".format(folder), "{0}_updated.yaml".format(file)), "w"
+        ) as write_file:
+            write_file.write(file_updated)
+    except Exception as e:
+        LOGGER.error(
+            "Error occurred while updating {0}.yaml : {1}".format(
+                os.path.join(folder, file), e
+            )
+        )
+
+
 @pytest.fixture(scope="session")
-def splunk_docker(
-    request, docker_services, docker_compose_files, tmp_path_factory, worker_id
-):
+def splunk_kubernetes(request):
     """
-    Splunk docker depends on lovely-pytest-docker to create the docker instance
+    Splunk kubernetes kubectl utiltity to create the kubernetes instance
     of Splunk this may be changed in the future.
-    docker-compose.yml in the project root must have
-    a service "splunk" exposing port 8000 and 8089
+    k8s_manifests in the project root must have
+    splunk_standalone consisting kubernetes manifests and shell script.
 
     Returns:
         dict: Details of the splunk instance including host, port, username & password.
     """
-    LOGGER.info("Starting docker_service=splunk")
-    if worker_id:
-        # get the temp directory shared by all workers
-        root_tmp_dir = tmp_path_factory.getbasetemp().parent
-        fn = root_tmp_dir / "pytest_docker"
-        with FileLock(str(fn) + ".lock"):
-            docker_services.start("splunk")
+    os.environ["TEST_RUNNER_DIRECTORY"] = os.getcwd()
+    parser = conf_parser.TABConfigParser()
+    parser.read(
+        os.path.join(
+            "{0}".format(os.getenv("SPLUNK_APP_PACKAGE")), "default", "app.conf"
+        )
+    )
+    splunk_addon_name = parser.get("package", "id")
+    try:
+        spl_path = os.path.join(
+            "{0}".format(os.getenv("TEST_RUNNER_DIRECTORY")), "tests", "src"
+        )
+        splunk_spl_name = [
+            filename
+            for filename in os.listdir(spl_path)
+            if filename.startswith(("{0}-").format(splunk_addon_name))
+        ]
+        SPLUNK_ADDON = str(splunk_spl_name[0])
+        LOGGER.info(SPLUNK_ADDON)
+    except Exception as e:
```

Comment:
Can we also check here version like `1.0.0` because it will also generate errors in tests right?

Reply:
Addon SPL will be of format `Addon_Name`-`version` ; version can be anything, we only require SPL with the given format, which will be installed while spinning up the splunk standalone instance.